### DISCUSSION TITLE
How to expose a datasource as API?

### DISCUSSION THREAD

**Discussion by thiagow** [Budibase/budibase#7288] [Help] [answered]
Is it possible to expose a data source created in Budibase as an API to be consumed by external applications via REST?

**Comment by andz-bb** [comment, marked as answer]
Hey, we have an API that should do what you're needing to do. The API reference with details of all of the available endpoints is here: https://docs.budibase.com/reference/post_applications

If you have any issues with it let me know and I'll be happy to help

**Reply by thiagow** [reply]
Hi Thomas, thanks for the reply. But I think I was not clear in my question.

My question is next. I'm creating a web system with Budibase, and I'll need to create a native app that will need to access the data that will be stored in the Budibase database. How could I expose this data to be accessed by API?

**Reply by andz-bb** [reply]
hey Thiago, sorry if I didn't explain it well earlier. We can use the Budibase API to query Budibase data sources using HTTP methods. For your request headers you would need your [Budibase API Key](https://docs.budibase.com/docs/public-api#how-to-get-your-api-key) for authorization and your [App ID](https://docs.budibase.com/docs/public-api#getting-an-app-id). These go in as the headers `x-budibase-api-key` and `x-budibase-app-id` respectively.

There's quite a few endpoints available depending on what you need to do with your data in your application. You have basic CRUD functionality available for rows on a given table. I would guess these are the endpoints that are most relevant for your use case so I'll give a really quick example of getting data from my Budibase DB using a post request.

I'm using the `rows/search` endpoint - you can see the [API reference for that here.](https://docs.budibase.com/reference/post_tables-tableid-rows-search) For me on a self hosted instance of Budibase I'm sending a POST to localhost:10000/api/public/v1/tables/[_my table Id]_/rows/search. The request headers have the two keys mentioned above and the url includes the table ID of the Budibase DB table I am searching. Then in the request body I am passing in an object with the parameters for the query. In this example, I am querying my City table to return all cities in Ireland. I'm using postman to simulate sending the HTTP requests from another app.

Here's my request headers with the API key and app ID:
![request headers](https://user-images.githubusercontent.com/110921612/184928017-14c62b37-74fd-4be2-8a91-8418ac5c346d.png)

My request object based on the API reference with the query I want to run on my table:
![request object](https://user-images.githubusercontent.com/110921612/184928129-b0e14928-8da6-4d53-a2d1-f7242589dd2d.png)

The response JSON from the API:
![response](https://user-images.githubusercontent.com/110921612/184928173-e547c347-8a69-404b-b6d2-41edfe7e3ec9.png)

Apologies if I'm still misunderstanding what you are trying to do here, but hopefully this helps! Let me know if this is useful for you 😊

**Reply by thiagow** [reply]
Thank you so much!

**Comment by tocasoft** [comment]
Hi. I can use the Budibase API to query Budibase data sources using HTTP methods without Budibase API Key ?
